src/shims/unix/freebsd/futex.rs

```diff
@@ -0,0 +1,67 @@
+use crate::helpers::EvalContextExt as HEvalContext;
+use crate::sync::EvalContextExt as SEvalContext;
+use crate::thread::EvalContextExt as TEvalContext;
+use crate::InterpResult;
+use crate::OpTy;
+use crate::{MiriEvalContext, Tag};
+
+enum OpType {
+    UmtxOpWait = 2,
+    //TODO: other types
+}
+
+impl TryFrom<i32> for OpType {
+    type Error = &'static str;
+
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        match value {
+            2 => Ok(OpType::UmtxOpWait),
+            _ => Err("Unsupported Futex operation"),
+        }
+    }
+}
+
+pub fn futex<'tcx>(
+    this: &mut MiriEvalContext<'_, 'tcx>,
+    obj: &OpTy<'tcx, Tag>,
+    op: i32,
+    val: u64,
+    uaddr: &OpTy<'tcx, Tag>,
+    uaddr2: &OpTy<'tcx, Tag>,
```

Comment:
Use the `Pointer` type for these two and move the conversion to the caller

Reply:
Needs some fixing

Reply:
ok, they are unused for now anyway, so let's just ignore them (keep OpTy and we'll see about them once they are used)